### DISCUSSION TITLE
`execute_in_process` does not produce run records on Dagit UI

### DISCUSSION THREAD

**Discussion by andger** [dagster-io/dagster#12069] [Q&A] [answered]
When calling `some_job.execute_in_process(..)`, it does not produce run records on Dagit UI. Is there a way to execute a job, which would produce a run record?

Thanks in advance

**Comment by schrockn** [comment, marked as answer]
@andger  

Thanks for your question. By default `execute_in_process` does not write to a persistent Dagster instance. This is deliberate, as it is designed for unit testing. However we do support this use case. Set the `instance` parameter to `DagsterInstance.get` and you should be good to go as long as the `DAGSTER_HOME` env var is set in your shell.

e.g: 

`some_job.execute_in_process(instance=DagsterInstance.get())`

**Comment by andger** [comment]
Thanks for the prompt answer @schrockn, I've already tried that. Still no run record produced. Here's a simple code snippet.
```
import random

from dagster import op, job, get_dagster_logger, make_values_resource, DagsterInstance


@op(required_resource_keys={"values"})
def log_resource(context):
    get_dagster_logger().info(context.resources.values)


def res():
    return {"values": make_values_resource(foo=int)}


@job(resource_defs=res())
def job_a():
    log_resource()


@op
def execute_a_job():
    job_a.execute_in_process(
        instance=DagsterInstance.get(),
        run_config={"resources": {"values": {"config": {"foo": random.randint(0, 10)}}}}
    )

@job
def job_b():
    execute_a_job()
```
`job_b` has a step that executes `job_a`, in the screenshot it executes successfully
![image](https://user-images.githubusercontent.com/18362280/216642921-e92d9316-af66-4c39-b0e0-e3a8b2d0893a.png)

But in the `job_a` runs tab there is no record of running it
![image](https://user-images.githubusercontent.com/18362280/216643028-864571c2-dd54-4a4f-a3e1-05bbbc4377d2.png)

Dagster version: 1.1.9

What we have now is a job that has a rather complicated resource definition and we want to have a zero-config option to execute it. We could you use `Field.default_value` to resolve the resources but then, in case we want to re-execute it for some reason, we cannot just simply go to previous job and press "Open in Launchpad". The resources that came through default variable do not carry over.

So we came up with an option to have a another "bootstrapping" job which has the capability to fill in all of the necessary resources. But the issues that the only way we're able to get to Run's to appear is either via `DagsterGraphQLClient` or by having a sensor which monitors the "bootstrapping" job and returns a `RunRequest`. Both way's seems to be quite hack-ish 😅

**Reply by alangenfeld** [reply]
> But in the job_a runs tab there is no record of running it

Did you check the full runs page? I think the issue is the scoping rules for the job_a runs page don't work with the direct in process targeting of the job since it lacks the context of what repository it is in.

You can use `metadata` to create a link to the downstream run in the event logs of the parent run. You can also use the `instance` off the `context` object directly. Something like this:

```
@op
def execute_a_job(context: OpExecutionContext):
    result = job_a.execute_in_process(
        instance=context.instance,
        run_config={"resources": {"values": {"config": {"foo": random.randint(0, 10)}}}}
    )
   context.add_output_metadata({"run": MetadataValue.dagster_run(result.run_id)})
 ```

**Reply by andger** [reply]
Yeah, it's in the full runs page... Haven't occurred to me to look there 🤦‍♂️ Thanks a lot, really helpful additional information.
